platform/darwin/src/MGLStyleValue.mm

```diff
@@ -63,6 +63,7 @@ - (instancetype)init {
 
 - (instancetype)initWithBase:(CGFloat)base stops:(NSDictionary *)stops {
     if (self = [super init]) {
+        NSAssert(stops.allKeys.count, @"%@ need at least one stop.", [self class]);
```

Comment:
For internal errors, use `NSAssert()`. For developer errors, raise an NSException and give it a name like `NSInvalidArgumentException` that the developer can identify the exception by.